src/main/java/ultoi/util/Parser.java

```diff
@@ -0,0 +1,127 @@
+package ultoi.util;
+
+import ultoi.task.Event;
+import ultoi.task.Deadline;
+
+import ultoi.command.Command;
+import ultoi.command.AddCommand;
+import ultoi.command.ByeCommand;
+import ultoi.command.DeleteCommand;
+import ultoi.command.FindCommand;
+import ultoi.command.ListCommand;
+import ultoi.command.MarkCommand;
+
+/**
+ * Represents a parser used to convert user input into valid commands.
+ *
+ * @author snoidetx
+ * @version 0.0.0
+ */
+public class Parser {
+    /**
+     * Returns a ultoi.command.Command object that input represents.
+     *
+     * @param input User input.
+     * @return ultoi.command.Command object that input represents.
+     * @throws UltoiException If input is not identifiable.
+     */
+    public static Command parse(String input) throws UltoiException {
+        String[] tokens = input.split(" ");
+
+        try {
+            switch (tokens[0]) {
+            case "list":
+                return new ListCommand(input);
```

Comment:
maybe you can consider adding `break;` to the end of each case or the comment `//Fallthrough`